zipline/pipeline/loaders/blaze/_core.pyx

```diff
@@ -352,7 +351,7 @@ cdef _array_for_column_impl(object dtype,
                 # upper bound doesn't include the timestamp because we've
                 # already included the timestamp-date in the baseline.
                 end = max(ts_ix - 1, 0)
-                if end >= asof_ix and out_array[asof_ix, column_ix] != value:
+                if end >= asof_ix: # and out_array[asof_ix, column_ix] != value:
```

Comment:
Do we still want this branch commented here?  What did the perf impact of this end up being?